connect.sh

```diff
@@ -43,3 +43,4 @@ sudo snap connect ${SNAP_NAME}:network-observe     :network-observe
 sudo snap connect ${SNAP_NAME}:process-control     :process-control
 sudo snap connect ${SNAP_NAME}:shutdown            :shutdown
 sudo snap connect ${SNAP_NAME}:dbus-wpa            wpa-supplicant:service
+sudo snap connect ${SNAP_NAME}:home                :home
```

Comment:
Can the certificates be stored in $SNAP_COMMON instead?

Reply:
They can be, sure.  They can also be stored in SNAP_DATA.  But $HOME is easier because it doesn't require sudo to move it into place.

Reply:
According to this: https://snapcraft.io/docs/snapcraft-interfaces, "Developers should endeavour to list only the plugs required for normal operation of the application."  I think that since :home is not required for normal operation, we should remove it.

Reply:
[We're already plugging the home interface](https://github.com/PelionIoT/snap-pelion-edge/blob/df8cee937a3e930d1684df8fc2f1a5fde4f3b583/snap/snapcraft.yaml#L26).  The change here adds it to the connect script used in developer mode and allows edge-proxy to access it.  I think users may want to store their certs in the $HOME directory, like maybe $HOME/ssl, especially since /etc/ssl is not writable in UC16.

Reply:
I think that plugging into home for ease of use would go against the guidance, since it's not a required plug.

Reply:
@moonchen This isn't just ease of use, it's a valid place to install SSL certs.  Can you guarantee that no customer would ever want to install certs into the HOME directory, especially since /etc/ssl is not writable?  This seems a valid use case to me.  Do you have some documentation on where the customer is expected to place the certs?

Reply:
snap provides $SNAP_COMMON for per-snap data, and $SNAP_USER_COMMON for per-user per-snap data.  Since the certificate is only used for this snap, I think one of those would be the correct place to store it.

Reply:
I'm interpreting the guideline to mean that if it's possible to use the snap without the interface, then the interface should not be added.  Regarding the interface already being in the snap, it looks like we might use home for docker.  If that's the case, then the change to add it to `connect.sh` would be long to another PR.

Reply:
It's possible to forcefully copy files into a snap's private area, yes.  But honestly is that the preferred way?  How do you propose that users (or devs too) provide certs to edge-proxy in both scenarios of provisioning/testing and normal runtime.

Reply:
What do you think about providing certs to edge-proxy like this:

```
chmod 400 proxy.crt proxy.key
sudo chown root:root proxy.crt proxy.key
sudo mv proxy.crt proxy.key /var/snap/pelion-edge/common/var/run/
snap set pelion-edge https-tunnel-tls-cert=/var/snap/pelion-edge/common/var/run/proxy.crt
snap set pelion-edge https-tunnel-tls-key=/var/snap/pelion-edge/common/var/run/proxy.key
```

Daemons run as root, and [root user can't bypass file permissions](https://forum.snapcraft.io/t/snapped-daemon-running-as-root-cannot-create-file-in-directory-with-odd-ownerships-permissions/17742) in snap.

Reply:
Here is the line at issue.  Copying files directly into the snap's private filesystem is problematic IMO.  A better solution is to copy it to HOME, i.e., /root in the case of daemons, or add code to the snap to generate the cert & key or fetch it in some way.
```
sudo mv proxy.crt proxy.key /var/snap/pelion-edge/common/var/run/
```

Reply:
In the interest of time, let's merge this for now and we can discuss possibly removing the home interface in a separate PR.